src/com/facebook/buck/apple/AppleDescriptions.java

```diff
@@ -311,6 +312,14 @@ public static void populateCxxLibraryDescriptionArg(
         SourceList.ofNamedSources(
             convertAppleHeadersToPublicCxxHeaders(
                 buildTarget, resolver::getRelativePath, headerPathPrefix, arg)));
+    if (arg.isModular()) {
+      output.addCompilerFlags(
+          StringWithMacros.of(
+              ImmutableList.of(
+                  Either.ofLeft(
+                      "-fmodule-name="
+                          + arg.getHeaderPathPrefix().orElse(buildTarget.getShortName())))));
```

Comment:
@robbertvanginkel Doesn't `AppleNativeTargetDescriptionArg` have `moduleName`? I feel like this logic of determining an effective module name is duplicated in quite a few places.